core/src/main/java/greencity/controller/TipsAndTricksController.java

```diff
@@ -0,0 +1,149 @@
+package greencity.controller;
+
+import greencity.annotations.ApiPageable;
+import greencity.constant.HttpStatuses;
+import greencity.dto.PageableDto;
+import greencity.dto.tipsandtricks.AddTipsAndTricksDtoRequest;
+import greencity.dto.tipsandtricks.AddTipsAndTricksDtoResponse;
+import greencity.dto.tipsandtricks.TipsAndTricksDto;
+import greencity.entity.TipsAndTricks;
+import greencity.service.TipsAndTricksService;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.security.Principal;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import springfox.documentation.annotations.ApiIgnore;
+
+@Validated
```

Comment:
I can't find which parameters are you validating. Please, remove this annotation if it is not needed

Reply:
Added parameter validation on the TipsAndTricksDtoRequest (validation of 'source' field, validation if provided tags are unique and if their quantity doesn't exceed the limit (3))